flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/DispatcherJob.java

```diff
@@ -119,11 +119,17 @@ private DispatcherJob(
 						}
 					});
 				} else { // failure during initialization
-					final Throwable strippedThrowable = ExceptionUtils.stripCompletionException(throwable);
+					JobStatus finalStatus = JobStatus.FAILED;
+					Throwable strippedThrowable = ExceptionUtils.stripCompletionException(throwable);
+					if (throwable instanceof JobInitializationCancelledException) {
```

Comment:
but there is no jobManagerRunner available if it's initialization failed exceptionally.

I will add a unit test that ensures the correct exception types to avoid accidental changes.

Reply:
I meant `this.initializingJobManagerRunner`.

Reply:
I see. I will look into it!